hie-base/Haskell/Ide/Engine/SemanticTypes.hs

```diff
@@ -62,6 +62,18 @@ data ModuleList = ModuleList {
   } deriving (Show,Read,Eq,Ord,Generic)
 
 -- ---------------------------------------------------------------------
+
+-- | GHC AST
+data AST = AST {
+    astModule :: T.Text
+  , astParsed      :: Value
+  , astRenamed     :: Value
```

Comment:
`Maybe Value`s are annoying since both `Nothing` and `Just Null` get serialized to `Null`. Also the `Null` in the `Value` makes the Nothing redundant at least as long as we don’t need two different null cases (I don’t know if that’s the case here).